aten/src/ATen/core/Generator.h

```diff
@@ -93,6 +93,9 @@ struct TORCH_API Generator {
   }
 
   void set_current_seed(uint64_t seed) { impl_->set_current_seed(seed); }
+  // Sets the offset of Generator state to the desired offset. This is currently
+  // supported for only Philox based Generators, i.e., CUDA and MPS.
+  void set_offset(uint64_t offset) { impl_->set_offset(offset); }
```

Comment:
It's a bit annoying that you are putting a subclass specific method un the base class